api/v1alpha1/ext_auth_types.go

```diff
@@ -63,6 +63,65 @@ type ExtAuth struct {
 	//
 	// +optional
 	RecomputeRoute *bool `json:"recomputeRoute,omitempty"`
+
+	// ContextExtensions are analogous to http_request.headers, however these
+	// contents will not be sent to the upstream server. This provides an
+	// extension mechanism for sending additional information to the auth server
+	// without modifying the proto definition. It maps to the internal opaque
+	// context in the filter chain.
+	//
+	// +optional
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=name
+	ContextExtensions []*ContextExtension `json:"contextExtensions,omitempty"`
+}
+
+// ContextExtensionValueType defines the types of values for ContextExtension supported by Envoy Gateway.
+//
+// +kubebuilder:validation:Enum=Inline;ValueRef
+type ContextExtensionValueType string
+
+const (
+	// ContextExtensionValueTypeInline defines the "Inline" ContextExtension type.
+	ContextExtensionValueTypeInline ContextExtensionValueType = "Inline"
+
+	// ContextExtensionValueTypeValueRef defines the "ValueRef" ContextExtension type.
+	ContextExtensionValueTypeValueRef ContextExtensionValueType = "ValueRef"
+)
+
+// ContextExtension is analogous to http_request.headers, however these
+// contents will not be sent to the upstream server. This provides an
+// extension mechanism for sending additional information to the auth server
+// without modifying the proto definition. It maps to the internal opaque
+// context in the filter chain.
+//
+// +kubebuilder:validation:XValidation:rule="(self.type == 'Inline' && has(self.inline) && !has(self.valueRef)) || (self.type == 'ValueRef' && !has(self.inline) && has(self.valueRef))",message="Exactly one of inline or valueRef must be set with correct type."
+type ContextExtension struct {
+	// Name of the context extension.
+	Name string `json:"name"`
+
+	// Type is the type of method to use to read the ContextExtension value.
+	// Valid values are Inline and ValueRef, default is Inline.
+	//
+	// +kubebuilder:default=Inline
+	// +unionDiscriminator
+	// +required
+	Type ContextExtensionValueType `json:"type"`
+
+	// Value of the context extension.
+	//
+	// +optional
+	// +unionMember
+	Inline string `json:"inline"`
```

Comment:
should be ptr with a omitempty tag since its optional

Reply:
IMO "inline" makes sense for a Lua script, because "value" does not. But for context extension, I would personally prefer "value" like [`gatewayv1.HTTPHeader`](https://pkg.go.dev/sigs.k8s.io/gateway-api@v1.4.0/apis/v1#HTTPHeader) or [`corev1.EnvVar`](https://pkg.go.dev/k8s.io/kubernetes/pkg/apis/core@v1.35.0#EnvVar)